plugin/src/main/java/org/opensearch/ml/sdkclient/DDBOpenSearchClient.java

```diff
@@ -219,20 +236,34 @@ public CompletionStage<UpdateDataObjectResponse> updateDataObjectAsync(UpdateDat
                     .tableName(getTableName(request.index()))
                     .key(updateKey)
                     .attributeUpdates(updateAttributeValue);
+                updateItemRequestBuilder
+                    .updateExpression("SET #seqNo = #seqNo + :incr")
+                    .expressionAttributeNames(Map.of("#seqNo", SEQ_NO_KEY))
+                    .expressionAttributeValues(Map.of(":incr", AttributeValue.builder().n("1").build()));
```

Comment:
Again, pretty sure this should be 0.  See [here](https://github.com/opensearch-project/OpenSearch/blob/77a74e2fcf2c6b56b0959a563fbc6a0d7aff220a/server/src/main/java/org/opensearch/index/seqno/SequenceNumbers.java#L52-L55); `NO_OPS_PERFORMED` is -1 so the initial indexing operation increments to 0.

Reply:
Ah, got it, this line of code says "increment by one".  

Would have helped with the classic needless comment:
```java
// increment n by one
n++;
```
😆